### PR TITLE
Key changes to CropXYZ function calls and events

### PR DESCRIPTION
**Pull request by ponyjackal**
Resolve OU-2666
Resolve OU-2667

### CODE REVIEW

**Comment by pcowgill** [comment]
@ponyjackal Could you take a look at the build failure?

File: src/contextualizers/protocol/index.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ import { bnsContextualizer } from './bns';
 import { highlightContextualizer } from './highlight';
 import { warpcastContextualizer } from './warpcast';
 import { boomboxContextualizer } from './boombox';
-import { goldContextualizer } from './gold';
+import { goldContextualizer } from './cropXYZ';
```
Comment (pcowgill) [inline src/contextualizers/protocol/index.ts:23]:
```suggestion
import { cropXyzContextualizer } from './cropXYZ';
```

File: src/types/contextAction/protocolContextAction.ts [hunk 2]
```diff
@@ -230,11 +230,17 @@ export type BoomboxContextAction =
 export enum GoldContextActionEnum {
```
Comment (pcowgill) [inline src/types/contextAction/protocolContextAction.ts:230]:
```suggestion
export enum CropXyzContextActionEnum {
```

File: src/types/contextAction/protocolContextAction.ts [hunk 3]
```diff
@@ -230,11 +230,17 @@ export type BoomboxContextAction =
...
   ACTIVATED_A_STARTER_PACK = 'ACTIVATED_A_STARTER_PACK',
   RECEIVED = 'RECEIVED',
+  HARVESTED_PLOT = 'HARVESTED_PLOT',
+  CLEARED_HARVEST = 'CLEARED_HARVEST',
+  STAKED_CROP = 'STAKED_CROP',
 }
 
 export type GoldContextAction =
```
Comment (pcowgill) [inline src/types/contextAction/protocolContextAction.ts:238]:
```suggestion
export type CropXyzContextAction =
```

File: src/contextualizers/protocol/cropXYZ/source.spec.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { Transaction } from '../../../types';
+import { detect, generate } from './source';
+import { containsBigInt, contextSummary } from '../../../helpers/utils';
+import goldSource0x31893393 from '../../test/transactions/goldSource-0x31893393.json';
+
+describe('Gold Source', () => {
```
Comment (pcowgill) [inline src/contextualizers/protocol/cropXYZ/source.spec.ts:6]:
```suggestion
describe('CropXYZ Source', () => {
```

File: src/contextualizers/protocol/cropXYZ/source.spec.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      goldSource0x31893393 as unknown as Transaction,
+    );
+    expect(isGoldSource1).toBe(true);
+  });
+
+  it('Should generate context', () => {
+    const transaction1 = generate(
+      goldSource0x31893393 as unknown as Transaction,
+    );
+    expect(transaction1.context?.summaries?.en.title).toBe('Gold');
```
Comment (pcowgill) [inline src/contextualizers/protocol/cropXYZ/source.spec.ts:18]:
```suggestion
    expect(transaction1.context?.summaries?.en.title).toBe('CropXYZ');
```

File: src/contextualizers/protocol/cropXYZ/source.spec.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { Transaction } from '../../../types';
+import { detect, generate } from './source';
+import { containsBigInt, contextSummary } from '../../../helpers/utils';
+import goldSource0x31893393 from '../../test/transactions/goldSource-0x31893393.json';
+
+describe('Gold Source', () => {
+  it('Should detect transaction', () => {
+    const isGoldSource1 = detect(
```
Comment (pcowgill) [inline src/contextualizers/protocol/cropXYZ/source.spec.ts:8]:
```suggestion
    const isCropXyzSource1 = detect(
```

File: src/contextualizers/protocol/cropXYZ/plotAction.ts [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    if (decoded && decoded.eventName === 'StakedCrop') {
+      const player = decoded.args['player'];
+      const plotId = decoded.args['plotId'];
+      transaction.context = {
+        summaries: {
+          category: 'PROTOCOL_1',
+          en: {
+            title: `Gold`,
```
Comment (pcowgill) [inline src/contextualizers/protocol/cropXYZ/plotAction.ts:131]:
```suggestion
            title: `CropXYZ`,
```

File: src/contextualizers/protocol/cropXYZ/plotAction.ts [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+    if (decoded && decoded.eventName === 'ClearedHarvest') {
+      const player = decoded.args['player'];
+      const plotId = decoded.args['plotId'];
+      transaction.context = {
+        summaries: {
+          category: 'PROTOCOL_1',
+          en: {
+            title: `Gold`,
+            default: '[[player]][[clearedHarvest]]plots[[plotId]]',
```
Comment (pcowgill) [inline src/contextualizers/protocol/cropXYZ/plotAction.ts:103]: We'll keep this like this for now, but an open question is whether we want to render the player given their non-trivial use of account abstraction, etc.

File: src/contextualizers/protocol/cropXYZ/destination.ts [hunk 9]
```diff
@@ -98,32 +97,38 @@ export function generate(transaction: Transaction): Transaction {
...
         break;
     }
   }
+
   if (
     !activatedStarterPackOnDestinationDecoded ||
-    !mintedPlotPackActivateDecoded ||
-    gameMintedTokenDecoded.length !== 2
+    !mintedPlotPackActivateDecoded
+    // || gameMintedTokenDecoded.length !== 2
```
Comment (pcowgill) [inline src/contextualizers/protocol/cropXYZ/destination.ts:111]: @ponyjackal What's the context on why this is commented out?

File: src/contextualizers/protocol/cropXYZ/plotAction.spec.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    expect(isPlotAction1).toBe(true);
+  });
+
+  it('Should generate context', () => {
+    const transaction1 = generate(
+      cropXyzPlotAction0x98d62314 as unknown as Transaction,
+    );
+    expect(transaction1.context?.summaries?.en.title).toBe('Gold');
+    expect(contextSummary(transaction1.context)).toBe(
+      '0x9a37e57d177c5ff8817b55da36f2a2b3532cde3f ACTIVATED_A_STARTER_PACK',
```
Comment (pcowgill) [inline src/contextualizers/protocol/cropXYZ/plotAction.spec.ts:20]: @ponyjackal We'll change this to be a different action, right?

File: src/contextualizers/protocol/cropXYZ/plotAction.ts [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+
+    if (decoded && decoded.eventName === 'HarvestedPlot') {
+      const player = decoded.args['player'];
+      const plotId = decoded.args['plotId'];
+      transaction.context = {
+        summaries: {
+          category: 'PROTOCOL_1',
+          en: {
+            title: `Gold`,
+            default: '[[player]][[harvested]]plots[[plotId]]',
```
Comment (pcowgill) [inline src/contextualizers/protocol/cropXYZ/plotAction.ts:74]: @ponyjackal We should use net asset transfers to detect the amount of tokens that moved to add to this

**Review by pcowgill — CHANGES_REQUESTED** [review]
Heading in the right direction! I requested a few changes in the comments.

File: src/contextualizers/protocol/cropXYZ/plotAction.ts (hunk 11 above)
Comment (pcowgill) [reply src/contextualizers/protocol/cropXYZ/plotAction.ts:74]: We can use the asset transfer helper in this repo to include this info

File: src/contextualizers/protocol/cropXYZ/destination.ts (hunk 9 above)
Comment (ponyjackal) [reply src/contextualizers/protocol/cropXYZ/destination.ts:111]: @pcowgill I cant not recall this correctly, but for the current contract, there is no event emitted for GameTokenMinted

File: src/contextualizers/protocol/cropXYZ/plotAction.spec.ts (hunk 10 above)
Comment (ponyjackal) [reply src/contextualizers/protocol/cropXYZ/plotAction.spec.ts:20]: Yeah, this test is not working, I tried to grab the only transaction in plotAction contract tx history, but it didnt work

Comment (pcowgill) [reply src/contextualizers/protocol/cropXYZ/plotAction.spec.ts]: Ah ok, but with the tx links I sent over in the Linear issue, you have txs you can use now, right?

Comment (ponyjackal) [reply src/contextualizers/protocol/cropXYZ/plotAction.spec.ts]: Yeah

File: src/contextualizers/protocol/cropXyz/plotAction.spec.ts [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+      '0x402533d5240a0c51e02d3714c0d0a057384d4872 HARVESTED_PLOT 257 0x9c82ca3332898bea9c9fa5f9642ba4a4628e1321',
+    );
+    expect(containsBigInt(transaction1.context)).toBe(false);
+
+    // const transaction2 = generate(
+    //   cropXyzPlotAction0x9bb5a737 as unknown as Transaction,
+    // );
+    // expect(transaction2.context?.summaries?.en.title).toBe('CropXYZ');
+    // expect(contextSummary(transaction2.context)).toBe(
+    //   '0x402533d5240A0c51e02d3714C0D0A057384D4872 HARVESTED_PLOT 257 0x9c82ca3332898bea9c9fa5f9642ba4a4628e1321',
```
Comment (pcowgill) [inline src/contextualizers/protocol/cropXyz/plotAction.spec.ts:41]:
```suggestion
    //   '0x402533d5240A0c51e02d3714C0D0A057384D4872 CLEARED_HARVEST for plot 257',
```

**Review by pcowgill — APPROVED** [review]

**Pull request merged**